### ISSUE TITLE
Arrow is unable to parse localized strings due to too strict regexp match on month tokens

### ISSUE THREAD

**Issue by swistakm** [crsmithdev/arrow#228]
Example:

```
In [1]: arrow.get('mar, 28 Abr 2015 22:13:04 -0500', 'ddd, D MMM YYYY HH:mm:ss Z' , locale='es')
---------------------------------------------------------------------------
ParserError                               Traceback (most recent call last)
(...)
ParserError: Failed to match token 'MMM' when parsing 'mar, 28 Abr 2015 22:13:04 -0500'
```

This is caused by [this part of code](https://github.com/crsmithdev/arrow/blob/master/arrow/parser.py#L33) that restricts parsed tokens to only those that could be found in `calendar` module:

```
_INPUT_RE_MAP = {
    # too strict month token patterns
    'MMMM': re.compile('({0})'.format('|'.join(calendar.month_name[1:])), re.IGNORECASE),
    'MMM': re.compile('({0})'.format('|'.join(calendar.month_abbr[1:])), re.IGNORECASE),
    # too strict meridians are also affected by this
    'a': re.compile('(a|A|p|P)'),
    'A': re.compile('(am|AM|pm|PM)'),
}
```

Interesting thing is that later in code locale is respected during parsing but only for months (not meridians):

```
elif token in ['MMMM', 'MMM']:
    parts['month'] = self.locale.month_number(value.capitalize())
```

So I suspect that this somewhere in past worked by there is an regression. This can be fixed in two ways:
1.  By using more wide matches on MMM and MMMM tokens like `(\w+)`. But this is more tricky than it seems to be because every language has different form of month abbreviations. For example many asians locale have months abbreviated as numbers with leading spaces and russian locale abbreviates months with dots.
2. By creating new _INPUT_RE_MAP attribute for every new instance of parser. So this no longer can be a class attribute and rexexp matches for those tokens would be populated with data from locale.

Problem with meridians is similar but additionally parser does not utilise locale when parsing them. But since this both issues are very related they could be fixed at the same time.

If you tell me how would you prefer to fix this problem I can take care of that and create pull request with fixes.

**Comment by beenje** [comment]
The month part is fixed by https://github.com/crsmithdev/arrow/commit/cb9acae33fbb49debb489014cda034fa9f6c437f

Same thing should be done for meridians.
Note that in the documentation, "A" is for "AM, PM" and "a" for "am, pm".
This is not what is currently implemented here.

**Comment by andrewelkins** [comment]
@swistakm If you can fix meridians in the same fashion in a PR that would be great!

**Comment by swistakm** [comment]
OK, I will make pull request. But which approach to take? Wider regexp matches (1) or instantiate `_INPUT_RE_MAP` on parser's `__init__()` call with data from locale (2)? I think second option would be better.

**Comment by swistakm** [comment]
I think this can be closed now since #244 was already merged long time ago and everything seems to work in latest version available on PyPI.
